### ISSUE TITLE
Suggested UX improvements for mini-timeline in `live` view

### ISSUE THREAD

**Issue by bagobones** [dermotduffy/frigate-hass-card#1055]
**Checklist:**

- [ x] I updated to the latest version available
- [x ] I cleared the cache of my browser

**Release with the issue:**
5 beta 1
**Last working release (if known):**
N/A
**Browser and Operating System:**
windows/chrome, iOS/Safari

**Description of problem:**
In the release notes mini timeline is shown with a vertical line showing current time and scrolling updating playback. I am not seeing this happen under live view. Also taping on events highlights them but it does not start playing them.

If I tap on the text for the HOUR it appears it is trying to load but the mini timeline disappears and the video window just goes blank.

My general expectation would be:
- dragging backward would skip to clip
- tapping on a event box would play the clip
- tapping on an hour or blank space would play recordings 

**Additional information:**
YAML config, using go2rtc as the live provider, however I did try removing that at setting it to a HA camera entities and didn't see any change.
```
type: custom:frigate-card
cameras:
  - frigate:
      camera_name: birdseye
  - frigate:
      camera_name: doorbell
    live_provider: go2rtc
  - frigate:
      camera_name: main_hall
    live_provider: go2rtc
  - frigate:
      camera_name: garage_front
    live_provider: go2rtc
  - frigate:
      camera_name: backyard
    live_provider: go2rtc
  - frigate:
      camera_name: living_room
    live_provider: go2rtc
  - frigate:
      camera_name: kitchen
    live_provider: go2rtc
  - frigate:
      camera_name: basement
    live_provider: go2rtc
  - frigate:
      camera_name: basement_hall
    live_provider: go2rtc
timeline:
  media: clips
media_gallery:
  controls:
    thumbnails:
      show_details: true
dimensions:
  aspect_ratio_mode: unconstrained
live:
  draggable: false
  controls:
    next_previous:
      style: none
    title:
      mode: popup-top-left
    timeline:
      mode: below
      media: clips
  transition_effect: none
  show_image_during_load: false
  preload: false
  control s:
    timeline:
      clustering_threshold: 2
menu: {}
```

**Comment by dermotduffy** [comment]
> I am not seeing this happen under live view.

This is working as intended. We cannot 'seek' the live view from the timeline, as we cannot understand the relationship between the time in the live feed (seconds into the video stream, which may just just literally started) and the time on the timeline, i.e. we cannot go arbitrarily forwards/backwards.

This is unlike media viewing, where we _can_ know this relationship because Frigate exports timestamped recording segments for recorded media.

> Also taping on events highlights them but it does not start playing them.

This is working as _planned_, but maybe should not be. If the thumbnail carousel is underneath (vs in the drawer) I think this makes a little more sense, since you see the thumbnail carousel move. But perhaps this is more intuitive to just immediately play the media if clicked in the mini-timeline in the live view.

Would you agree?

**Comment by bagobones** [comment]
Hmm, again using other security products as an example you can scrub the timeline back and it will just start playing the closest clip to that time stamp (there will be an indicator that you are no longer viewing live and there is a button to jump back to live view.

At the very least I expected the timeline to be an alternative to the left write drawer OR carousel allowing an instant hop to playback mode with the timeline showing where you are or which event is playing.

IE tap on an event box and that event plays, tap on a shaded part of the hour and the recording plays, or tap on the time, and it start playing recordings starting at that hour.

After turning on other options it appears to act as a filter to drawer or carousel which is interesting but not really intuitive.

**Comment by dermotduffy** [comment]
There are at least two things getting mixed up here which makes this challenging to discuss / understand. I am going to try to keep them separate here.

#### 1. What should happen, in live mode, when an event in the mini-timeline is clicked.
**Current**: Currently the card will choose the event in the thumbnail carousel and nothing else.
**Your proposal**: Immediately play the chosen media, leaving the `live` view.

Correct? This is straightforward to implement.

#### 2. What should happen, in live mode, when an timeline is dragged.
**Current**: It alllows the user to pan/zoom back in time to see the timeline of events and do $something (as defined by 1 above) when clicked on.
**Your proposal**: I'm not sure: I think you are suggesting leaving `live` view entirely if I drag the timeline, change to the media viewer and start playing the "nearest" event media instead to where the user dragged.

There are hard parts here: Lets say in `live` view I have the timeline configured to be above the card, and in the media viewer I have the timeline configured to be below the card. What would you expect to happen in this case when the user drags the timeline on the live view?

**Comment by bagobones** [comment]
> #### 1. What should happen, in live mode, when an event in the mini-timeline is clicked.
> **Current**: Currently the card will choose the event in the thumbnail carousel and nothing else. **Your proposal**: Immediately play the chosen media, leaving the `live` view.
> 
> Correct? This is straightforward to implement.

Correct

I find the current selection confusing as it isn't even visible that it has happened if you have the drawer on the left or the right closed. Also could not immediately figure out how to clear it. I will try it with the carousel enabled.

> #### 2. What should happen, in live mode, when the mini-timeline is dragged.
> **Current**: It alllows the user to pan/zoom back in time to see the timeline of events and do $something (as defined by 1 above) when clicked on. **Your proposal**: I'm not sure: I think you are suggesting leaving `live` view entirely if I drag the timeline, change to the media viewer and start playing the "nearest" event media instead to where the user dragged.
> 
> There are hard parts here: Lets say in `live` view I have the timeline configured to be above the card, and in the media viewer I have the timeline configured to be below the card. What would you expect to happen in this case when the user drags the timeline on the live view? This is perhaps a silly example, I'm just point out this may not be especially seamless.

Yes that is what I am proposing. (also how surveillance station and a few other products handle things, just filling to playback or live mode with a button)

I think the issue is that while I understand that live and media viewer are separate as coded currently, visually as a user I don't see why they are treated as separate. IE simplicity from the users point of view I have a "video window" and a timeline or list of events. I want to be able to jump between live and the past events quickly.. It doesn't matter to me that the player is technically something completely different as long  as when I click the live icon it shows me live, and if I click on the drawer / carousel /  mini timeline it start playing the past and clearly shows that it is doing so.

I guess the simplest way to put it is that I think the live viewer and media viewer should be merged, they have highly redundant visual elements and configuration items, the only technical difference I can see is that one is showing live view port and the other is showing a recorded video with playback timeline.

**Comment by bagobones** [comment]
FYI did a bit of testing with the bottom carousel mode AND the mini timeline open at the same time and it does make WAY more sense now, however I still think if timeline is enabled on its own it should be triggering playback  or switch to media playback view.

**Comment by dermotduffy** [comment]
> I think the issue is that while I understand that live and media viewer are separate as coded currently, visually as a user I don't see why they are treated as separate. IE simplicity from the users point of view I have a "video window" and a timeline or list of events. I want to be able to jump between live and the past events quickly.. It doesn't matter to me that the player is technically something completely different as long as when I click the live icon it shows me live, and if I click on the drawer / carousel / mini timeline it start playing the past and clearly shows that it is doing so.

Absolutely understand, and that is exactly how you should think about it! I'm just pointing out that there are users who likely have a different configuration settings that the card already supports where they will expect the card to behave in potentially a different ways -- and changes we make may break that for them. Which also might be fine, just need to be principled :-) 

I'll have a play around and see what I can do here.

> FYI did a bit of testing with the bottom carousel mode AND the mini timeline open at the same time and it does make WAY more sense now, however I still think if timeline is enabled on its own it should be triggering playback or switch to media playback view.

Figured it would! I think my options here are either (1) Just change it to always play, or (2) Change it to always play if the thumbnail carousel is either off or in the drawer. Will think on it.

**Comment by bagobones** [comment]
Maybe this should be a separate item but for the single camera / mini time line it would be nice if all events could be crushed into one line vertically .. I have tried playing with the clustering setting but I always end up with some vertically offset events when they are close together. Maybe? Clustering isn't working as well on the mini timeline? I have it set to 2 so it should be clustering almost everything should it not?

**Comment by dermotduffy** [comment]
Quick update on this: There are two main parts:

* Changing the behavior of "click" on the timeline [**done**].
* Changing the behavior of panning the timeline into or out of the past.

The second one is harder than I thought (at least to make it a semi-smooth) experience -- but I'm working  on it ...

**Comment by bagobones** [comment]
> * Changing the behavior of panning the timeline into or out of the past.
> 
> The second one is harder than I thought (at least to make it a semi-smooth) experience -- but I'm working on it ...

Visually this Axis app is a good example of my "ideal" mini timeline. https://www.youtube.com/watch?v=qSSCOq_5Ov0 
1. Single event ribbon showing activity
2. Clear virtical indicator of "now" or the moment playing that advances with playback
3. Jumps to clips as you go back in time.. (doesn't need to be as instant just that if you find a spot and linger for a moment it should jump to that clip / point in time.

Going to say I am REALLY loving v5, these are all just suggestions.. However this STYLE of mini timeline is similar among many of the top camera / doorbell / NVR interfaces.

**Comment by dermotduffy** [comment]
To set your expectations: This issue is exclusively going to cover the [two points mentioned here](https://github.com/dermotduffy/frigate-hass-card/issues/1055#issuecomment-1501161825).

**Comment by dermotduffy** [comment]
I'm going to close this for now as the two points are implemented. It's a pretty big change to the structure of the DOM though, so I would not be surprised if some unintended display bug creeps in. 

However panning smoothly between the 'past' and 'live' is pretty fun :-)

@bagobones If you'd like to try it out, you can either (i) wait for the next beta release, (ii) build it yourself, or (iii) use the build artifacts here: https://github.com/dermotduffy/frigate-hass-card/suites/12123427668/artifacts/639026018

Would love feedback on it via issue reports if something seems broken (please keep it to strictly one "topic" per issue report though).

Thank you for the continued feedback!

**Comment by bagobones** [comment]
I will wait for the next beta but thank-you for all your hard work.

**Comment by felipecrs** [comment]
@dermotduffy is there a chance you can record a video showing how it works? I'm a little confused on how to set this up.

**Comment by dermotduffy** [comment]
It's no big deal really. You just turn on the timeline for live & media_viewer and then you can just scan back and forth between the "past" and the "present".

https://user-images.githubusercontent.com/1136829/230818864-f3cd17e0-4ff4-4bfe-9491-ff86b37e2797.mp4

The capture didn't grab the tooltips, but clicking the little icon on the bottom right lets you choose the 'pan' behavior:

* [Default] Panning seeks between all media.
* Panning seeks across the select event only (useful for crowed events).
* Panning just pans the timeline.

**Comment by felipecrs** [comment]
Wow that's interesting! Thanks a lot!

**Comment by bagobones** [comment]
Just loaded beta2 and this is awesome..

I can see I can force live mode when switching cameras by changing the defaults but is there a shortcut to jumping back to live mode on the current camera if you have scrolled WAY back already other than scrolling too or past the present time?

**Comment by dermotduffy** [comment]
> is there a shortcut to jumping back to live mode on the current camera if you have scrolled WAY back already other than scrolling too or past the present time?

Can't you just press the live button in the menu?

**Comment by bagobones** [comment]
I had to think long and hard about wha relive button you where talking about.

Till this moment I have never need to tap on that camera icon in the main menu.

**Comment by mdelpire** [comment]
I tested the 5.0.0 beta 2 and it works great. The only thing I feel annoying with the timeline below the live feed is that if I click and hold to move the timeline to the left or right, it directly switch to the recording view at the timestamp I clicked, while I only wanted to move the timeline to see another point in time. In the real timeline icon, I don't have this behaviour, only with the live view.
